From c8cca54b8e84e901ae36b0b9dd746a23dd62739e Mon Sep 17 00:00:00 2001
From: Ian Campbell <ian.campbell@citrix.com>
Date: Thu, 19 Aug 2010 11:56:59 +0100
Subject: [PATCH 03/26] xen: use dynamic_irq_init_keep_chip_data

This synchronises with what went upstream in:
commit 99ad198c4978036bb9f7ebd11618b225b77046da
Author: Stefano Stabellini <stefano.stabellini@eu.citrix.com>
Date:   Fri May 14 12:41:20 2010 +0100

    xen: Fix find_unbound_irq in presence of ioapic irqs.

Signed-off-by: Ian Campbell <ian.campbell@citrix.com>
Cc: Stefano Stabellini <stefano.stabellini@eu.citrix.com>
(cherry picked from commit 9fa90aa72d6af5cc2c2eddf56f9a586035e13ae7)

Signed-off-by: Ian Campbell <ijc@hellion.org.uk>
---
 drivers/xen/events.c |    6 +-----
 1 files changed, 1 insertions(+), 5 deletions(-)

diff --git a/drivers/xen/events.c b/drivers/xen/events.c
index f5cfa2b..181a254 100644
--- a/drivers/xen/events.c
+++ b/drivers/xen/events.c
@@ -377,7 +377,6 @@ static int find_unbound_irq(void)
 	int irq;
 	struct irq_desc *desc;
 	int start = get_nr_hw_irqs();
-	void *chip_data;
 
 	if (start == nr_irqs)
 		goto no_irqs;
@@ -403,10 +402,7 @@ static int find_unbound_irq(void)
 	if (WARN_ON(desc == NULL))
 		return -1;
 
-	/* save and restore chip_data */
-	chip_data = desc->chip_data;
-	dynamic_irq_init(irq);
-	desc->chip_data = chip_data;
+	dynamic_irq_init_keep_chip_data(irq);
 
 	return irq;
 
-- 
1.7.5.4

